File: 409-longest-palindrome/longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char,int>m;
        for(int i=0;i<s.size();i++)m[s[i]]++;

        int cnt=0;
        // int maxi=0;
        for(auto it:m){
            if(it.second%2==0)cnt+=it.second;
            if(it.second%2!=0)cnt+=(it.second-1);
            // if(it.second%2!=0)maxi=max(maxi,it.second);
            }

        cnt+=1;
        if(cnt>s.size())return s.size();
                return cnt;
    }
};

File: 409-longest-palindrome/longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
         int fl=0;
         Map<Character,Integer>m=new HashMap();
         for(char ch:s.toCharArray()){
           m.put(ch,m.getOrDefault(ch,0)+1);
         }
         int res=0;

         for(char k:m.keySet()){
            res+=(m.get(k)/2)*2;
            if(m.get(k)%2==1)fl=1;
         }

         if(fl==1)res+=1;

         return res;
    }
}